stmhal/can.c

```diff
@@ -525,6 +556,52 @@ STATIC mp_obj_t pyb_can_setfilter(mp_uint_t n_args, const mp_obj_t *pos_args, mp
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_KW(pyb_can_setfilter_obj, 1, pyb_can_setfilter);
 
+STATIC mp_obj_t pyb_can_rxcallback(mp_obj_t self_in, mp_obj_t fifo_in, mp_obj_t callback_in) {
+    pyb_can_obj_t *self = self_in;
+    mp_int_t fifo = mp_obj_get_int(fifo_in);
+
+
+    if (callback_in == mp_const_none){
+        if (fifo == 0){
+            __HAL_CAN_DISABLE_IT(&self->can, CAN_IT_FMP0);
+            self->rxcallback0 = mp_const_none;
+        } else if (fifo == 1){
+            __HAL_CAN_DISABLE_IT(&self->can, CAN_IT_FMP1);
+            self->rxcallback1 = mp_const_none;
+        }
+    } else if (mp_obj_is_callable(callback_in)) {
+        if (self->can_id == PYB_CAN_1){
+            if (fifo == 0) {
+                self->rxcallback0 = callback_in;
```

Comment:
I guess it is a matter of style. Personally I think this way of writing code is a bit easier to read an understand, but I will not make any noise if you want me to change it before accepting the code.